src/utils/contractreflectioninterface.h

```diff
@@ -0,0 +1,917 @@
+     TypeToEnum<const std::vector<Address> &>::value},
+    {meta::resolve_type<bool>(), TypeToEnum<bool>::value},
+    {meta::resolve_type<bool &>(), TypeToEnum<bool &>::value},
+    {meta::resolve_type<const bool &>(), TypeToEnum<const bool &>::value},
+    {meta::resolve_type<std::vector<bool>>(),
+     TypeToEnum<std::vector<bool>>::value},
+    {meta::resolve_type<std::vector<bool> &>(),
+     TypeToEnum<std::vector<bool> &>::value},
+    {meta::resolve_type<const std::vector<bool> &>(),
+     TypeToEnum<const std::vector<bool> &>::value},
+    {meta::resolve_type<std::string>(), TypeToEnum<std::string>::value},
+    {meta::resolve_type<const std::string &>(),
+     TypeToEnum<const std::string &>::value},
+    {meta::resolve_type<std::string &>(), TypeToEnum<std::string &>::value},
+    {meta::resolve_type<std::vector<std::string>>(),
+     TypeToEnum<std::vector<std::string>>::value},
+    {meta::resolve_type<std::vector<std::string> &>(),
+     TypeToEnum<std::vector<std::string> &>::value},
+    {meta::resolve_type<const std::vector<std::string> &>(),
+     TypeToEnum<const std::vector<std::string> &>::value}};
+
+/**
+ * This function returns the ABI type string for a given ABI type.
+ * @param type The ABI type.
+ * @return The ABI type string.
+ */
+std::string inline getStringFromABIEnum(ABI::Types type) {
+  const std::unordered_map<ABI::Types, std::string> typeMappings = {
+    {ABI::Types::uint256, "uint256"},
+    {ABI::Types::uint256Arr, "uint256[]"},
+    {ABI::Types::address, "address"},
+    {ABI::Types::addressArr, "address[]"},
+    {ABI::Types::boolean, "bool"},
+    {ABI::Types::booleanArr, "bool[]"},
+    {ABI::Types::bytes, "bytes"},
+    {ABI::Types::bytesArr, "bytes[]"},
+    {ABI::Types::string, "string"},
+    {ABI::Types::stringArr, "string[]"}
+  };
+
+  auto it = typeMappings.find(type);
+  if (it != typeMappings.end()) {
+    return it->second;
+  } else {
+    throw std::runtime_error("Unsupported ABI type");
+  }
+}
+
+/**
+ * This function returns the ABI type for a given ABI type string.
+ * @param type The ABI type string.
+ * @return The ABI type.
+ */
+ABI::Types inline getABIEnumFromString(const std::string& type) {
+  static const std::unordered_map<std::string, ABI::Types> typeMappings = {
+    {"uint256", ABI::Types::uint256},
+    {"uint256[]", ABI::Types::uint256Arr},
+    {"address", ABI::Types::address},
+    {"address[]", ABI::Types::addressArr},
+    {"bool", ABI::Types::boolean},
+    {"bool[]", ABI::Types::booleanArr},
+    {"bytes", ABI::Types::bytes},
+    {"bytes[]", ABI::Types::bytesArr},
+    {"string", ABI::Types::string},
+    {"string[]", ABI::Types::stringArr}
+  };
+
+  auto it = typeMappings.find(type);
+  if (it != typeMappings.end()) {
+    return it->second;
+  } else {
+    throw std::runtime_error("Invalid type");
+  }
+}
+
+static std::unordered_map<std::string, std::vector<std::string>>
+    constructorArgumentNamesMap; /// Map to store constructor argument names
```

Comment:
Doxygen might not understand the line comments like that, append an "<" (e.g. `///< Map to...`).

Reply:
Done!